rest-api/src/main/java/io/quarkus/qe/RepositoryResource.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.qe;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.qe.model.Repository;
+import io.quarkus.qe.services.RepositoryService;
+
+@Path("/repository")
+@Transactional
+public class RepositoryResource {
+
+    @Inject
+    RepositoryService repositoryService;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/{id}")
+    public Repository get(@PathParam("id") Long id) {
+        return repositoryService.findById(id);
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public void add(@Valid Repository request) {
+        repositoryService.sendNewRepositoryRequest(request);
```

Comment:
This should return an HTTP 202. And If  you make, for example, let's say...10 consecutive requests and doesn't really make any impact (are idempotent) then the HTTP method should be PUT.

Reply:
I agree with the two points you commented. I've already changed the status to return 202. But I didn't change the method to PUT as I added a new change to return a HTTP Conflict if the repository URL already exists, therefore now this endpoint is not idempotent any longer.
Let me know your thoughts about the new changes. Thanks!